src/components/drawer/drawer.pw.tsx

```diff
@@ -0,0 +1,488 @@
+
+    await expect(drawerSidebarContentInnerElement(page, 0)).toHaveText(
+      "link a"
+    );
+    await expect(drawerSidebarContentInnerElement(page, 0)).not.toBeVisible();
+  });
+
+  test("should render component opened when the expanded prop is true", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<DrawerCustom expanded />);
+
+    await expect(drawerSidebarContentInnerElement(page, 0)).toHaveText(
+      "link a"
+    );
+    await expect(drawerSidebarContentInnerElement(page, 0)).toBeVisible();
+  });
+
+  test("should render component with custom sidebar", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<DrawerCustomSidebar />);
+
+    await expect(drawerSidebarContentInnerElement(page, 0)).toHaveText(
+      "cypress"
+    );
+    await expect(drawerSidebarContentInnerElement(page, 0)).toBeVisible();
+  });
+
+  test("should render component with custom expandedWidth", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<DrawerCustom expandedWidth="65%" />);
+
+    await assertCssValueIsApproximately(
+      drawer(page).locator("div").nth(0),
+      "width",
+      887
+    );
+  });
+
+  test("should render component with custom backgroundColor", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<DrawerCustom backgroundColor="#FF0000" />);
+
+    const color = "rgb(255, 0, 0)";
+
+    await expect(drawer(page).locator("div").nth(0)).toHaveCSS(
+      "background-color",
+      color
+    );
+    await expect(drawer(page).locator("div").nth(0)).toHaveCSS(
+      "border-right-color",
+      color
+    );
+  });
+
+  test("should render component with custom height", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<DrawerCustom height="75%" />);
+
+    await expect(drawer(page)).toHaveAttribute("height", "75%");
+  });
+
+  test("should render component with custom title", async ({ mount, page }) => {
+    await mount(<DrawerCustom title="cypress_title" />);
+
+    await expect(drawer(page).locator("div").locator("div").nth(0)).toHaveText(
+      "cypress_title"
+    );
+  });
+
+  test("should render component with toggle control when showControls prop is true", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<DrawerCustom showControls />);
+
+    await expect(drawerToggle(page)).toBeVisible();
+  });
+
+  test("should render component without toggle control when showControls prop is falsy", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<DrawerCustom />);
+
+    expect(await drawerToggle(page).count()).toEqual(0);
+  });
+
+  test("should render component with custom footer", async ({
+    mount,
+    page,
+  }) => {
+    await mount(
+      <div
+        style={{
+          height: "400px",
+        }}
+      >
+        <DrawerCustomFooterHeader />
+      </div>
+    );
+
+    const footer = stickyFooter(page);
+    await footer.scrollIntoViewIfNeeded();
+    expect(await stickyFooter(page).count()).toBeGreaterThan(0);
```

Comment:
**comment** (non-blocking): I don't think this `count` test is needed - it won't pass `toBeVisible()` below if it's not actually found in the DOM!